app/domain/operations/event_logs/create.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require "dry/monads"
+require "dry/monads/do"
+
+module Operations
+  module EventLogs
+    # Create Event Log entity
+    class Create
+      send(:include, Dry::Monads[:result, :do])
+
+      attr_accessor :resource_class
+
+      # @param [Hash] opts Options to build audit log event
+      # @option opts [<GlobalID>] :subject_gid required
+      # @option opts [<String>]   :correlation_id required
+      # @option opts [<Symbol>]   :event_category required
+      # @option opts [<String>]   :session_id optional
+      # @option opts [<String>]   :account_id optional
+      # @option opts [<String>]   :host_id required
+      # @option opts [<String>]   :trigger required
+      # @option opts [<String>]   :response required
+      # @option opts [<Symbol>]   :log_level optional
+      # @option opts [<Symbol>]   :severity optional
+      # @option opts [<DateTime>] :event_time required
+      # @option opts [<Array>]    :tags optional
+      # @return [Dry::Monad] result
+      def call(params)
+        values = yield validate(params)
+        entity = yield create(values)
+
+        Success(entity)
+      end
+
+      private
+
+      def validate(params)
+        @resource_class = params.delete(:resource_class)
+        return Failure("resource class name missing") unless resource_class
+
+        result =
+          "AcaEntities::EventLogs::#{resource_class}EventLogContract"
+          .constantize
+          .new
```

Comment:
this is unsafe, we need to fix this security issue; my proposal will be to write a factory that doesn't receive the class name directly and we get it indirectly
https://stackoverflow.com/questions/37364454/what-is-a-ruby-factory-method

Reply:
We can also use the case statement.

```
case resource_class
when 'Osse'
  AcaEntities::EventLogs::OsseEventLogContract
when 'HbxEnrollment'
  AcaEntities::EventLogs::HbxEnrollmentEventLogContract
when 'Audit'
  AcaEntities::EventLogs::AuditEventLogContract
when 'Login'
  AcaEntities::EventLogs::LoginEventLogContract
else
  nil
end
```